docs/fastify.asciidoc

```diff
@@ -0,0 +1,204 @@
+[[fastify]]
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/nodejs/current/fastify.html[elastic.co]
+endif::[]
+
+== Get started with Fastify
+
+Getting Elastic APM set up for your Fastify app is easy,
+and there are various ways you can tweak it to fit your needs.
+
+Follow the guide below and refer to the <<api,API documentation>> for all the advanced stuff.
+
+[float]
+[[fastify-installation]]
+=== Installation
+
+Add the `elastic-apm-node` module as a dependency to your application:
+
+[source,bash]
+----
+npm install elastic-apm-node --save
+----
+
+[float]
+[[fastify-initialization]]
+=== Initialization
+
+It's important that the agent is started before you require *any* other modules in your Node.js application - i.e. before `fastify`, `http`, etc.
+
+This means that you should probably require and start the agent in your application's main file (usually `index.js`, `server.js` or `app.js`).
+
+Here's a simple Fastify example with the Elastic APM agent installed:
+
+[source,js]
+----
+// Add this to the VERY top of the first file loaded in your app
+var apm = require('elastic-apm-node').start({
```

Comment:
I would also suggest to add `apm` to the Fastify instance by using the [`decorate`](https://www.fastify.io/docs/latest/Decorators/) API, in this way you can make `apm` accessible in every part of the application, which is very useful if you need to catch custom errors (as you have documented below).

```js
const apm = require('elastic-apm-node').start({ ... })

const fastify = require('fastify')()
fastify.decorate('apm', apm) // now you can do `fastify.apm` to access it
...

```

Reply:
The node agent is a singleton, and every time you `require('elastic-apm-node)` we return the agent to you. You of course only need to call `start()` once, which you usually do in you index file. So you can just require the agent again in another file to get access to it. That's what you meant right?

Reply:
> That's what you meant right?

Yes! But since the agent is a singleton, you don't need to use the `decorate` API :)